static/templates/topic_sidebar_actions.handlebars

```diff
@@ -31,5 +31,12 @@
         </a>
     </li>
 
+    <li>
+        <a class="sidebar-popover-delete-topic-messages" data-stream-id="{{ stream_id }}" data-topic-name="{{ topic_name }}">
+            <i class="fa fa-trash" aria-hidden="true"></i>
+            {{#tr this}}Delete all messages in <b>__topic_name__</b>{{/tr}}
+        </a>
+    </li>
```

Comment:
This should have a confirmation dialogue similar to the one for deleting a single message, with text "Delete all messages in topic __topic_name__?".